core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -659,6 +662,48 @@ class LogManager(logDirs: Seq[File],
       Option(currentLogs.get(topicPartition))
   }
 
+  /**
+   * Method to indicate that logs are getting initialized for the partition passed in as argument.
+   * This method should always be followed by [[kafka.log.LogManager#finishedInitializingLog]] to indicate that log
+   * initialization is done.
+   */
+  def initializingLog(topicPartition: TopicPartition): Unit = {
+    partitionsInitializing(topicPartition) = false
+  }
+
+  /**
+   * Mark the partition configuration for partitions that are getting initialized for topic
+   * as dirty. That will result in reloading of configuration once initialization is done.
+   */
+  def topicConfigUpdated(topic: String): Unit = {
+    partitionsInitializing.collect {
+      case (topicPartition, _) if topicPartition.topic() == topic => partitionsInitializing(topicPartition) = true
+    }
+  }
+
+  /**
+   * Mark all in progress partitions having dirty configuration if broker configuration is updated.
+   */
+  def brokerConfigUpdated(): Unit = {
+    partitionsInitializing.foreach {
+      case (topicPartition, _) => partitionsInitializing(topicPartition) = true
```

Comment:
We might need to be a little careful since we no longer have the lock. It seems possible that `finishedInitializingLog` could be called concurrently. In that case, should we use a method like `replace` so that we atomically update the value only if it its current value is `false`? Similarly in `topicConfigUpdated`.

Reply:
Thanks for catching this. Fixed.